parser-typechecker/src/Unison/PrintError.hs

```diff
@@ -629,51 +629,68 @@ renderTypeError e env src curPath = case e of
         _ -> Pr.wrap $ "It should be of type " <> Pr.group (style Type1 (renderType' env expectedType) <> ".")
   UnknownTerm {..} ->
     let (correct, wrongTypes, wrongNames) =
-          foldr sep id suggestions ([], [], [])
-        sep (C.Suggestion name typ _ match) r =
+          foldr sep id (sortOn (Text.length . C.suggestionName) suggestions) ([], [], [])
```

Comment:
Is this meant to sort by the length of the FQN?
If you're aiming to select the most relevant names you might want to count the number of `.` separated segments instead.